pom.xml

```diff
@@ -170,11 +170,13 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.11.0</version>
-          <!-- available from Java 9, but we went back to jdk8 for building
```

Comment:
@BalikhinLB Is this change related to the relevant one or could that be reverted?

Reply:
This change is required for the application to work.
inside parameterNameDiscoverer.getParameterNames(method):
```
if (!param.isNamePresent()) {
                return null;
}
```
  and this change need for param.isNamePresent() = true

Reply:
You can see it [here](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/StandardReflectionParameterNameDiscoverer.html)

Reply:
@berndruecker  I think this change is required. It seems the new `StandardReflectionParameterNameDiscoverer ` requires some additional compiler help . This [article](https://docs.oracle.com/javase/tutorial/reflect/member/methodparameterreflection.html) helps explains it.

My only concern would be, it seems due to this new required compiler flag, we would require our new and existing customers to add this new flag on their build. 

Our customers would use their own projects and create their own Zeebe Workers and they annotate it with `@Variable` for the method parameters. For all of this to work, the compiler flag needs to be enabled on their respective projects. I'll verify that, and see if thats the case. This would be a breaking change to them and docs updates might be needed.

Reply:
Yes I can confirm that if you are using Zeebe workers and utilizing parameter names such as `@Variable` then customers would be required to compile their code with `parameters`.  We gotta need to think more on this one regarding the user-experience. 

@BalikhinLB I think we can remove the configuration on the parent pom.xml, and just put it under the `spring-boot-starter-camunda-example`

![image](https://github.com/camunda-community-hub/spring-zeebe/assets/133918079/1cd69256-da67-4be3-a900-0700c1ee1904)

It looks like this is only required on code that uses the custom Zeebe worker annotations. Other modules doesn't need this.

Reply:
The ReadZeebeWorkerValueTest test is failing because it checks the MethodInfo from ClassInfoTest.WithZeebeWorker.class, and it is precisely here that parameters are needed. The JobHandlerTest, SmokeTest are also failing.
To avoid test failures, parameters need to be added in spring-client-zeebe, test/embedded, and test/testcontainer. If you think this is okay - I will do so. Honestly, I don't really understand why java.version can be 1.8 when spring now practically (except for the most ancient versions) only works on Java 11+ versions.

Reply:
@BalikhinLB Good catch. Yeah please apply the parameters to above affected modules.

Reply:
@1nb0und ok, i applied this

Reply:
I think this change is good to go in. Also updated readme in this PR https://github.com/camunda-community-hub/spring-zeebe/pull/514 so users are aware when using new Spring-Zeebe versions higher than 8.3.1

@berndruecker thoughts on this?

Reply:
Bummer that this is required to continue having the same functionality, as this is really a breaking change from user perspective. But I agree that this is probably OK to do now - also good to add it to the readme (I commented in #514 - I think we should specifically say that this is only required for parameter annotations to work).

I wonder if this is just a Spring related thing or if this is generally required to process parameter annotiations - but don't have time to dive into this and leave it to your judgement @1nb0und . 

In short: yeah - ok for me